### PR TITLE
feat(chat): support topics (Issue #2086)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

add topics

**Issues:**

- Issue #2086 
- 
**Design:**

https://www.figma.com/design/lwL5jneveZczAj5RLcLFH2/DIAL-Chat?node-id=10464-16326&t=BjnL9IGIAxHF8Pa1-1

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2269.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Common/ApplicationDialog.tsx [hunk 1]
```diff
@@ -432,6 +464,74 @@ const ApplicationDialogView: React.FC<Props> = ({
...
+                  }
+                  onGetOptionBorderColor={(value) =>
+                    getTopicColors(value, {
+                      defaultBorderColor: 'var(--bg-layer-4)',
+                    }).borderColor
+                  }
+                  options={[
+                    { value: 'dev', label: 'dev' },
+                    { value: 'dev1', label: 'dev1' },
+                  ]}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationDialog.tsx]: Do we need this? We should read options from config

File: apps/chat/src/utils/app/style-helpers.ts [hunk 2]
```diff
@@ -3,3 +3,17 @@ import { FocusEvent } from 'react';
...
+    defaultBgColor,
+    defaultBorderColor,
+  }: { defaultBgColor?: string; defaultBorderColor?: string } = {
+    defaultBgColor: 'var(--bg-accent-primary-alpha)',
+    defaultBorderColor: 'var(--stroke-accent-primary)',
+  },
+) => ({
+  backgroundColor: `var(--bg-topic-${topic.toLowerCase()}, ${defaultBgColor})`,
+  borderColor: `var(--stroke-topic-${topic.toLowerCase()}, ${defaultBorderColor})`,
+});
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/style-helpers.ts]:
```suggestion
export const getTopicColors = (topic: string) => ({
  backgroundColor: `var(--bg-topic-${topic.toLowerCase()}, var(--bg-topic-default, "#FALLBACK"))`,
  borderColor: `var(--stroke-topic-${topic.toLowerCase()}, var(--bg-stroke-default, "#FALLBACK"))`,
});
```

File: apps/chat/src/components/Common/ApplicationDialog.tsx (hunk 1 above)
Comment (Alexander-Kezik) [reply apps/chat/src/components/Common/ApplicationDialog.tsx]: This is just to represent that dropdown is working

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2269.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2269.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2269.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**